### ISSUE TITLE
VkQueue is simultaneously used by multiple threads

### ISSUE THREAD

**Issue by rolandhill** [vsg-dev/VulkanSceneGraph#892]
Hi Robert,

I get a message that vkQueue is being used simultaneously by multiple threads when resizing the window. I suspect that it only happens when I'm also using the DatabasePager. The message is (from vsgPagedLod --debug):

UNASSIGNED-Threading-MultipleThreads(ERROR / SPEC): msgNum: 337425955 - Validation Error: [ UNASSIGNED-Threading-MultipleThreads ] Object 0: handle = 0x560f83645bc0, type = VK_OBJECT_TYPE_QUEUE; | MessageID = 0x141cb623 | THREADING ERROR : vkQueueSubmit(): object of type VkQueue is simultaneously used in thread 0x7fe35ae60bc0 and thread 0x7fe33e7fe640
    Objects: 1
        [0] 0x560f83645bc0, type: 4, name: NULL

I get a lot of them in my own application that makes heavy use of the DatabasePager for tiled terrain pulled from disk. I've also managed to get it using **./vsgpagedlod --debug**, but it is harder to reproduce, possibly because the tiles are pulled more slowly from the internet. 

To reproduce, in vsgPagedLod --debug, zoom into the terrain so that it is gradually generating higher resolution tiles, then grab the corner of the window and constantly resize it. It took around 1 minute to generate an error for me (and takes about 1 second in my own application).

I'm wondering if objects being recompiled during a window resize are trying to push commands into a queue at the same time as database tiles are being compiled - just a guess.

I'm using Kubuntu 22.04 and an Nvidia 3070Ti card. VSG is master from July 18.

Thanks and regards,
Roland

**Comment by robertosfield** [comment]
Hi Roland,

The only places the VSG uses vkQueueSubmit is in src/vsg/Queue.cpp and in
both cases there is a mutex that prevents multi-threaded invocation on the
same queue:

https://github.com/vsg-dev/VulkanSceneGraph/blob/master/src/vsg/vk/Queue.cpp

This leaves the question of whether the debug layer is erroneous or whether
there is some other mechanism that is implicitly invoking a vkQueueSubmit
outside of vsg::Queue.

Could you look to see if there is a way of getting the stack trace for the
two threads that the debug layer is flagging?

Cheers,
Robert.

**Comment by rolandhill** [comment]
Hi Robert,

The program doesn't crash when it throws the error, so I haven't been able to get a stack trace at that point of time. I did print the 4 databasepager thread addresses and they don't match the thread addresses in the error message. In fact, the thread addresses in the error are a long way from any addresses I've seen in my application.

I've also updated the Vulkan SDK to the latest version (as well as VSG*) and the number of errors being thrown seams to be reduced. I'm starting to wonder if it isn't a false error from the debug layer as well. 

Leave it with me. I'll close this now and re-open it later if I need to.

Cheers,
Roland
